[Feature] Canned responses search - Matcher for non-adjacent keywords

Zaahed opened the pull request:
Resolves #5367 

This custom matcher will return results for non-adjacent keywords in the canned responses search field (ticket view). Makes searching for canned responses easier when you don't know the exact order of the words, and when you have a large list of canned responses.

Zaahed committed:
Canned responses search (ticket view) - Custom matcher for non-adjacent keywords

JediKev commented:
@Zaahed 

The problem with this is that the results are not sorted properly. With this applied, exact matches are on bottom where they should be the first results. I'm sure a lot of people will be upset that the exact matches are not priority in the result set (how it is currently). If you could find a way to sort the results properly I would see this as a legitimate pull.

Cheers.

Zaahed committed:
Canned responses search (ticket view) - Sorting results

Canned responses search (ticket view) - Removed unnecessary brackets if statement

Canned responses search (ticket view) - Rebase commit

Zaahed commented:
Hi @JediKev 

Thanks for your feedback. I've pushed a new commit which sorts the results and shows the exact matches at the top.

JediKev commented:
We should do this at database level instead of in JS.

Zaahed commented:
I assume this means we should use an AJAX request similar to that of the lookup function used in the ticket list search bar.

However this also means that there will be a small delay (150+ ms) because of the AJAX request, whereas the JS returns the results instantly at each keystroke.

corent56 commented:
Maybe it should be a good idea that when we search for a character like é or è, à... we can see all the answer with or without accent (e->e-é-è-ê ...)
